astroquery/eso/core.py

```diff
@@ -695,16 +729,63 @@ def retrieve_data(self, datasets, continuation=False, destination=None):
                     raise RemoteServiceError("There was a remote service "
                                              "error; perhaps the requested "
                                              "file could not be found?")
-            for fileId in root.select('input[name=fileId]'):
-                log.info("Downloading file {0}...".format(fileId.attrs['value'].split()[0]))
-                fileLink = ("http://dataportal.eso.org/dataPortal" +
-                            fileId.attrs['value'].split()[1])
+
+            if with_calib != 'none':
+                # when requested files with calibrations, some javascript is
+                # used to display the files, which prevent retrieving the files
+                # directly. So instead we retrieve the download script provided
+                # in the web page, and use it to extract the list of files.
+                # The benefit of this is also that in the download script the
+                # list of files is de-duplicated, whereas on the web page the
+                # calibration files would be duplicated for each exposure.
+                try:
+                    link = root.select('a[href$=/script]')[0]
+                    if 'downloadRequest' not in link.text:
+                        # Make sure that we found the correct link
+                        raise Exception
+                    href = link.attrs['href']
+                    script = self._request("GET", href, cache=False)
+                    fileLinks = re.findall(
+                        r'"(https://dataportal.eso.org/dataPortal/api/requests/.*)"',
+                        script.content.decode('utf8'))
+
+                    # links with api/ do not work, wtf ???
+                    fileLinks = [
+                        f.replace('https://dataportal.eso.org/dataPortal/api/requests',
+                                  'https://dataportal.eso.org/dataPortal/requests')
+                        for f in fileLinks]
+
+                    log.info("Detecting already downloaded datasets, "
+                             "including calibrations...")
+                    fileIds = [f.rsplit('/', maxsplit=1)[1] for f in fileLinks]
+                    filteredIds, files = self._check_existing_files(
+                        fileIds, continuation=continuation,
+                        destination=destination)
+
+                    fileLinks = [f for f, fileId in zip(fileLinks, fileIds)
+                                 if fileId in filteredIds]
+                except Exception:
```

Comment:
Also best to catch a specific exception class here

Reply:
generic Exception catching will also catch a poorly-timed ctrl-c

Reply:
This was really for development and because I was not sure if the solution was going to work. I can remove this I guess.
Btw, `Exception` does not catch keyboard interrupt, it's only bare except and `BaseException` who do.

Reply:
Oh, ok, I didn't realize that!  Thanks.